src/applications/personalization/profile-2/tests/e2e/profile.cypress.spec.js

```diff
@@ -0,0 +1,105 @@
+import { PROFILE_PATHS } from '../../constants';
+
+import mockUser from '../fixtures/users/user-36.json';
+import mockPaymentInfo from '../fixtures/payment-information/direct-deposit-is-set-up.json';
+import mockFeatureToggles from '../fixtures/feature-toggles.json';
+
+/**
+ *
+ * @param {boolean} mobile - test on a mobile viewport or not
+ *
+ * This helper:
+ * - loads the Profile, confirms the user is redirected to the correct URL, and
+ *   then performs an aXe scan
+ * - clicks through each item in the sub-nav, checks that the URL is correct,
+ *   and performs an aXe scan
+ */
+function checkAllPages(mobile = false) {
+  cy.visit('/profile');
+  if (mobile) {
+    cy.viewport('iphone-6');
+  }
+
+  // should show a loading indicator
+  cy.findByRole('progressbar').should('exist');
+  cy.findByText(/loading your information/i).should('exist');
+
+  // and then the loading indicator should be removed
+  cy.findByRole('progressbar').should('not.exist');
+  cy.findByText(/loading your information/i).should('not.exist');
+
+  // visiting /profile should redirect to profile/personal-information
+  cy.url().should(
+    'eq',
+    `${Cypress.config().baseUrl}${PROFILE_PATHS.PERSONAL_INFORMATION}`,
+  );
+
+  // make the a11y check on the Personal Info section
+  cy.injectAxe();
+  cy.checkA11y();
+
+  // make the  a11y check on the Military Info section
+  if (mobile) {
+    cy.findByRole('button', { name: /your profile menu/i }).click();
+  }
+  cy.findByRole('link', { name: 'Military information' }).click();
+  cy.url().should(
+    'eq',
+    `${Cypress.config().baseUrl}${PROFILE_PATHS.MILITARY_INFORMATION}`,
+  );
+  cy.checkA11y();
+
+  // make the a11y check on the Direct Deposit section
+  if (mobile) {
+    cy.findByRole('button', { name: /your profile menu/i }).click();
+  }
+  cy.findByRole('link', { name: 'Direct deposit' }).click();
+  cy.url().should(
+    'eq',
+    `${Cypress.config().baseUrl}${PROFILE_PATHS.DIRECT_DEPOSIT}`,
+  );
+  cy.checkA11y();
```

Comment:
What do you think about making another function out of this that gets called when iterating over something like:

```js
[
  { linkLabel: 'Direct deposit', profilePath: PROFILE_PATHS.DIRECT_DEPOSIT },
  { linkLabel: 'Account security', profilePath: PROFILE_PATHS.ACCOUNT_SECURITY },
  ...
].forEach(axeCheckSection);
```

Reply:
I like it. Maybe a little over-optimized but I like it 😀